apiserver/facades/client/bundle/mock_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bundle_test
+
+import (
+	"github.com/juju/testing"
+	"gopkg.in/juju/charm.v6"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/facades/client/bundle"
+	"github.com/juju/juju/state"
+	statetesting "github.com/juju/juju/state/testing"
+)
+
+type mockState struct {
+	testing.Stub
+	bundle.Backend
+	entities map[string]state.Entity
+	app      mockApplication
+	unit     mockUnit
+}
+
+func newMockState() *mockState {
+	unitsChanges := make(chan []string, 1)
```

Comment:
No need for any of these watchers etc - this mock state only needs to do one thing - provide the Export() method.

Reply:
done